### ISSUE TITLE
Honor queries with codes like zh_CN or pt_BR

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-server#4091]
The Android app is making queries like that return nothing:
https://ssl-api.openfoodfacts.org/api/v0/product/6924743919235.json?fields=ingredients_text_zh-CN%2Cproduct_name_zh-CN

On the other hand, this works.
https://ssl-api.openfoodfacts.org/api/v0/product/6924743919235.json?fields=ingredients_text_zh%2Cproduct_name_zh

Can we add Country removal to the API so that we can serve the values stored ?

pt-BR

**Comment by teolemon** [comment]
We already do it in write mode: 
https://github.com/openfoodfacts/openfoodfacts-server/blob/f62ffbab9c4cabc4be91f8f6b93919cc9ce02cd5/cgi/product_jqm_multilingual.pl#L216

**Comment by hangy** [comment]
IMHO, the app shouldn't be sending these requests right now. The `pt_BR` to `pt` change was implemented as a _workaround_ for some misbehaving external client, but we currently don't support more complex ISO codes officially (also see #667). Simply removing stuff could be really hurtful in cases like `zh-hans` vs `zh-hant` vs `zh-tw`.

**Comment by stephanegigandet** [comment]
I agree with @hangy, we don't want apps to send us extended language codes, the more workaround we had, the more bugs we will have down the road. Here the right thing to do is to change the OFF app so that it sends only the 2 letter code.

**Comment by teolemon** [comment]
I'll get the Android app fixed, and that way, you'll have no way to tell zh-hans vs zh-hant vs zh-tw apart, neither now, nor retroactively when we implement proper country support.
